### PR TITLE
refactor: consolidate prettier config

### PR DESCRIPTION
**Pull request by holic**
- moves all prettier usage to the root package
- adjusts prettier command to match lint-staged extensions (and runs prettier across repo)
- adds lint step to CI

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/** @type {import('prettier').Config} */
+module.exports = {
+  plugins: ["prettier-plugin-solidity"],
+  printWidth: 120,
+  semi: true,
+  tabWidth: 2,
+  useTabs: false,
+  bracketSpacing: true,
+};
```
Comment (holic) [inline .prettierrc.js:9]: moved this to a `.js` file so other packages can include/extend in the future with

```
/** @type {import('prettier').Config} */
module.exports = {
  ...require('../../.prettierrc.js'),
  // extended config values or overrides here
};
```

**Review by alvrs — APPROVED** [review]
very nice!

**Comment by holic** [comment]
unclear why prettier at the root is disagreeing with cli-generated output (which runs through the same prettier version using the same config + plugin)

@dk1a do you recall any issues with this?

**Comment by dk1a** [comment]
yes I remember it happening, but I hadn't really looked into the cause

**Comment by holic** [comment]
Oh these are a separate autogen pipeline (ejs -> sol, not using prettier), so that's why they differ.

**Pull request merged**